mlserver/codecs/numpy.py

```diff
@@ -92,7 +92,21 @@ def _encode_data(data: np.ndarray, datatype: str) -> list:
             # need to encapsulate it into a list so that it's compatible.
             return [data.tobytes()]
 
-    return data.flatten().tolist()
+    flattened_list = data.flatten().tolist()
+
+    # Replace NaN with null
+    has_nan = np.isnan(data).any()
+    if has_nan:
+        flattened_list = list(map(convert_nan, flattened_list))
+
+    return flattened_list
+
+
+def convert_nan(val):
```

Comment:
Given that this is used in both pandas and numpy codecs, does it make sense to be move out to the common codecs/util.py file :thinking:

Reply:
I can see that in the pandas codec files from numpy codec are being imported. Is this correct?

Reply:
It's quite specific to pandas / numpy, so I'd leave it as-is